tests/jax/core_system_components/config_test.py

```diff
@@ -295,3 +383,124 @@ def test_accidental_parameter_override_with_update_exception(
             param_0=2, param_2="param"
         )
         config.update(component_0=other_hyperparameter_config)
+
+
+def test_add_non_dataclass_config_exception(config: Config) -> None:
+    """Test that exception is thrown when when trying to add component config that \
+        is not a dataclass.
+
+    Args:
+        config : Mava config
+    """
+
+    non_dataclass_type_config_dict = {"name": "component", "setting": 5}
+
+    with pytest.raises(Exception):
+        config.add(component=non_dataclass_type_config_dict)
+
+
+def test_add_before_built_exception(
+    config: Config, dummy_component_config_0: HyperparameterConfig
+) -> None:
+    """Tests that exception is thrown when when trying to add component config that \
+        has already been built.
+
+    Args:
+        config : Mava config
+        dummy_component_config_0: component config dataclass
+    """
+
+    config.add(component_0=dummy_component_config_0)
+    config.build()
+
+    with pytest.raises(Exception):
+        config.add(component_1=dummy_component_config_0)
+
+
+def test_add_same_component_twice_exception(
+    config: Config, dummy_component_config_0: HyperparameterConfig
+) -> None:
+    """Test that exception is thrown when when trying to add component configs that \
+        have the same name.
+
+    Args:
+        config : Mava config
+        dummy_component_config_0: component config dataclass
+    """
+
+    config.add(component_same_name=dummy_component_config_0)
+
+    with pytest.raises(Exception):
+        config.add(component_same_name=dummy_component_config_0)
```

Comment:
Would be nice to test this with a component that isn't exactly the same, so add config_2 that has 1 attribute name the same as config_0 to test this

Reply:
thinking about it you could possibly use existing components to do this if you aren't adding them all to the same system already

Reply:
Good idea. Added an extra test case called `test_add_same_component_twice_exception_different_parameters`